test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -250,6 +250,9 @@ spec =
         Test.specify "methods of Integer" <|
             Meta.meta Integer . methods . sort . should_equal ['round', 'truncate']
 
+        Test.specify "static methods of Integer" <|
+            Meta.meta (Meta.type_of Integer) . methods . sort . should_equal ['parse', 'parse_builtin', 'round', 'truncate']
```

Comment:
I think ideally we should add a test for an "Atom" type, like `My_Type`.

We should make sure that this also works correctly for user-defined atom types.
`Integer` is a primitive, so it is a bit special. We had lots of inconsistencies between handling primitive values, polyglot values and atoms - so let's verify if we are consistent here.

Reply:
Done as described at https://github.com/enso-org/enso/pull/6938/files#r1215276648